app/services/scholars_archive/triple_powered_service.rb

```diff
@@ -0,0 +1,13 @@
+module ScholarsArchive
+  # Provide select options for the copyright status (edm:rights) field
+  class TriplePoweredService
+    def fetch(uris)
+      labels = []
+      uris.each do |uri|
+        graph = ScholarsArchive::TriplePoweredProperties::Triplestore.fetch(uri)
+        labels << ScholarsArchive::TriplePoweredProperties::Triplestore.predicate_labels(graph).values.flatten.compact
+      end
+      labels.flatten.compact
```

Comment:
this is returning all of the labels for all of the URIs, I think you would want to return a hash of an array of labels for each uri so that they can be indexed accordingly